sycl/doc/SharedLibraries.md

```diff
@@ -359,6 +375,25 @@ of defined symbols. If this assumption is not correct, there can be two cases:
 So, it is valid to pick the met first device image which defines required symbol
 during search.
 
+##### Function pointers support
+
+In addition to imported and exported symbols, information about indirectly
+called functions also used during search of device images.
+
+When the program manager gets a request to create a program object using device
+image, it examines its property that contains info about signatures of
+functions that were used via function pointers or marked with
+`indirectly_callable` attribute.
+If it is not empty, then for each described function signature, program manager
+finds all device images that have a record with the same function signature.
+In other words, all device images that record uses of pointers to a function
+with the same signature must be linked together.
```

Comment:
Will this lead to increase of JIT compilation time and make results of device code split to naught?

Reply:
It can, but there is no other way to guarantee accessibility of `indirectly_callable` function to all its possible indirect callers.

Reply:
Another my question requires an example.
Lets have a kernel `A` which will indirectly call `i32 (i32)` function and `float ()` function.
The image with this kernel requires to be linked against images which declare to have `i32 (i32)` function and the other one which contains `float ()` function? Do we need to link against the third image if the second one contains functions of both signatures?

Reply:
The thing is you don't know which one of those two `float()` functions were used so you have to link both